### PR TITLE
SALTO-8930: Support Okta automations

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add support in Okta automation.
The type is not documented in Swagger file, but we can get it using policies endpoint (using `type=USER_LIFECYCLE`)

---

_Additional context for reviewer_

---
_Release Notes_: 

_Okta_adapter_:
- Support fetch and deploy of Okta automations

---
_User Notifications_: 

_Okta_adapter_:
- Automation and AutomationRule elements will be added

### CODE REVIEW

File: packages/okta-adapter/src/config.ts [hunk 1]
```diff
@@ -115,9 +115,14 @@ const POLICY_TYPE_NAME_TO_PARAMS: Record<PolicyTypeNames, PolicyParams> = {
...
   },
+  [AUTOMATION_TYPE_NAME]: {
+    queryParam: 'USER_LIFECYCLE',
+    ruleName: 'AutomationRule',
+    policyServiceUrl: '/admin/lifecycle-automation#tab-policy/{id}',
+  },
 }
 
-const getPolicyItemsName = (policyName: string): string => (`${(policyName).slice(0, -1)}ies`)
+const getPolicyItemsName = (policyName: string): string => (policyName === 'Automation' ? 'Automations' : `${(policyName).slice(0, -1)}ies`)
```
Comment (edenhassid) [inline packages/okta-adapter/src/config.ts:125]: why not use `AUTOMATION_TYPE_NAME`

File: packages/okta-adapter/src/filters/delete_fields.ts [hunk 2]
```diff
@@ -15,12 +15,13 @@
 */
 import { Element, isInstanceElement } from '@salto-io/adapter-api'
 import { FilterCreator } from '../filter'
-import { APPLICATION_TYPE_NAME, AUTHORIZATION_POLICY_RULE, GROUP_TYPE_NAME, POLICY_TYPE_NAMES } from '../constants'
+import { APPLICATION_TYPE_NAME, AUTHORIZATION_POLICY, GROUP_TYPE_NAME, POLICY_TYPE_NAMES, AUTOMATION_TYPE_NAME } from '../constants'
 
 const TYPES_TO_FIELDS: Record<string, string[]> = {
   [GROUP_TYPE_NAME]: ['roles'],
   [APPLICATION_TYPE_NAME]: ['appUserSchema'],
-  [AUTHORIZATION_POLICY_RULE]: ['policyRules'],
```
Comment (edenhassid) [inline packages/okta-adapter/src/filters/delete_fields.ts:23]: why did you remove this?

**Review by edenhassid — APPROVED** [review]
looks good! 🔥  added a small nit and a question

File: packages/okta-adapter/src/filters/delete_fields.ts (hunk 2 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/filters/delete_fields.ts:23]: just noticed I put the wrong type in here 😓  the change fixes this
